src/components/interface/Header/ToolTranslate.js

```diff
@@ -1,25 +1,28 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
 import '@gouvfr/dsfr/dist/component/translate/translate.css';
 
 import dataAttributes from '../../../utils/data-attributes';
 import typeValidation from '../../../utils/type-validation';
+import useOnClickOutside from '../../../hooks/useOnClickOutside';
 
 const ToolTranslate = ({
   children, currentLang, className, descCurrentLang, title, ...remainingProps
 }) => {
   const [isOpen, setIsOpen] = useState(false);
   const _className = classNames(className, 'fr-translate fr-nav');
+  const translateMenuRef = useRef();
+  useOnClickOutside(translateMenuRef, () => setIsOpen(false));
```

Comment:
Can you put the function in a mnemo or a callback? so it's not recreated everytime ?